### PR TITLE
feat(lighthouse-service): Run lighthouse-service without distributor sidecar

### LINKED ISSUE #7435: core: adapt internal services to use `cp-connector`
With https://github.com/keptn/keptn/pull/7418 `cp-connector` will be introduced to connect Keptn integrations/services to the control plane. The following services need to be adapted to make use `cp-connector` instead of the `distributor`:

- [ ] lighthouse-svc - POC available (1)
- [x] approval-svc
- [x] api-svc (this service only sends events to NATS)
- [ ] mongodb-datastore

## Acceptance Criteria
- [ ] All control plane services above are able to run **without** distributor but using the `cp-connector` library
- [ ] Helm charts are updated to don't use the distributor

## DoD 

- [ ] Manual teast of install/upgrade via helm w/ and w/o `--reuse-values` 
- [ ] Adapt the integration tests 

## Resources

(1) https://github.com/keptn/keptn/pull/7334

### PR DESCRIPTION
**Pull request by odubajDT**
Signed-off-by: odubajDT <ondrej.dubaj@dynatrace.com>

## This PR

- runs lighthouse-service without distributor using cp-connector library
- adapts changes in cp-connector library

### Related Issues

Fixes #7435 

### Integration tests
https://github.com/keptn/keptn/actions/runs/2313253710

### CODE REVIEW

**Review by RealAnna — APPROVED** [review]
LGTM

**Review by thisthat — COMMENTED** [review]
Why approval- and api-service are also touched?

**Comment by odubajDT** [comment]
> Why approval- and api-service are also touched?

update of cp-connector dependency to the newest version

File: lighthouse-service/event_handler/handler.go [hunk 1]
```diff
@@ -17,10 +19,11 @@ type EvaluationEventHandler interface {
 	HandleEvent(ctx context.Context) error
 }
 
-func NewEventHandler(event cloudevents.Event) (EvaluationEventHandler, error) {
+func NewEventHandler(ctx context.Context, event cloudevents.Event) (EvaluationEventHandler, error) {
 	logger.Debug("Received event: " + event.Type())
 
-	keptnHandler, err := keptnv2.NewKeptn(&event, keptncommon.KeptnOpts{})
+	eventSender := ctx.Value(controlplane.EventSenderKey).(controlplane.EventSender)
```
Comment (bacherfl) [inline lighthouse-service/event_handler/handler.go:25]: can we do an additional check if the cast was successful here?
Some thing like 

```
eventSender, ok := ctx.Value(controlplane.EventSenderKey).(controlplane.EventSender)
	if !ok {
		return nil, errors.New("could not get event sender from context")
	}
```

File: test/go-tests/test_uniform.go [hunk 2]
```diff
@@ -398,8 +401,9 @@ func Test_UniformRegistration_RegistrationOfKeptnIntegrationMultiplePods(t *test
 func Test_UniformRegistration_RegistrationOfKeptnIntegrationRemoteExecPlane(t *testing.T) {
 	// install echo integration
 	// make sure the echo-service uses the same distributor as Keptn core
-	distributorImage, err := GetImageOfDeploymentContainer("lighthouse-service", "distributor")
+	imageName, err := GetImageOfDeploymentContainer("lighthouse-service", "lighthouse-service")
 	require.Nil(t, err)
+	distributorImage := CreateDistributorImage(imageName, "lighthouse-service")
```
Comment (bacherfl) [inline test/go-tests/test_uniform.go:406]: I believe here we can just use `strings.Replace()` directly, as it's not immediately clear what the `CreateDistributorImage()` function does

**Review by bacherfl — APPROVED** [review]

**Pull request merged**